control.lua

```diff
@@ -156,165 +158,205 @@ end
+			return
 		end
-		if transfer_rate < VALID_POWER_TRANSFER_RATES[1] then
-			transfer_rate = VALID_POWER_TRANSFER_RATES[1]
+	end
+	
+	-- find the nearest connected power pole
+	local D = game.max_electric_pole_supply_area_distance + factory.layout.outside_size / 2
+	local canidates = {}
+	for _, entity in ipairs(surface.find_entities_filtered{type="electric-pole", area={{x-D, y-D}, {x+D,y+D}}}) do
+		if entity.electric_network_id == electric_network and entity ~= pole then
+			canidates[#canidates+1] = entity
 		end
 	end
-	factory.transfer_rate = transfer_rate
-	local power_string, transfer_text = "",""
-	if transfer_rate >= 1000 then
-		power_string = (transfer_rate / 1000) .. "GW"
-	else
-		power_string = transfer_rate .. "MW"
+	
+	if #canidates == 0 then return end
+	connect_power(factory, surface.get_closest({x, y}, canidates))
+end
+
+local function power_pole_placed(pole)
+	local D = pole.prototype.supply_area_distance + 10
+	local position = pole.position
+	local x = position.x
+	local y = position.y
+	
+	for _, entity in ipairs(pole.surface.find_entities_filtered{type=BUILDING_TYPE, area={{x-D, y-D}, {x+D,y+D}}}) do
+		if not HasLayout(entity.name) then goto continue end
+		factory = get_factory_by_building(entity)
+		local electric_network = factory.outside_energy_receiver.electric_network_id
+		if electric_network == nil or electric_network ~= pole.electric_network_id then goto continue end
+		if electric_network == factory.inside_power_poles[1].electric_network_id then goto continue end
+		
+		connect_power(factory, pole)
+		
+		::continue::
 	end
-	if positive then
-		transfer_text = "factory-connection-text.power-transfer-increased"
-	else
-		transfer_text = "factory-connection-text.power-transfer-decreased"
+end
+
+local function power_pole_destroyed(pole)
+	pole.disconnect_neighbour()
+	
+	for _, factory in pairs(global.factories) do
+		if factory.built and factory.outside_power_pole and factory.outside_power_pole.valid and factory.outside_power_pole == pole then
+			update_power_connection(factory, pole)
+		end
 	end
-	factory.inside_surface.create_entity{
-		name = "flying-text",
-		position = {x = factory.inside_x + factory.layout.energy_indicator_x, y = factory.inside_y + factory.layout.energy_indicator_y}, color = {r = 228/255, g = 236/255, b = 0},
-		text = {transfer_text, power_string},
-		force = factory.force
-	}
-	update_power_settings(factory)
+	
+	cleanup_middlemen()
 end
 
 -- FACTORY UPGRADES --
 
 local function build_lights_upgrade(factory)
 	if factory.upgrades.lights then return end
 	factory.upgrades.lights = true
-	for _, pos in pairs(factory.layout.lights) do
-		local light = factory.inside_surface.create_entity{name = "factory-ceiling-light", position = {factory.inside_x + pos.x, factory.inside_y + pos.y}, force = factory.force}
-		light.destructible = false
-		light.operable = false
-		light.rotatable = false
-		table.insert(factory.inside_other_entities, light)
-	end
+	factory.inside_surface.daytime = 1
```

Comment:
Wasn't this done so that solar panels don't produce while in the factories?
You can set `surface.solar_power_multiplier = 0` to make solar panels ineffective

Reply:
> Wasn't this done so that solar panels don't produce while in the factories? You can set `surface.solar_power_multiplier = 0` to make solar panels ineffective

surface.solar_power_multiplier does not seem to work for cross surface poles